src/groovy/org/pih/warehouse/api/StockMovement.groovy

```diff
@@ -230,8 +232,8 @@ class StockMovement {
         if (electronicType) {
             User user = AuthService.currentUser.get()
             def accessRule = ConfigHelper.findAccessRule("stockRequest", "remove")
-            def userRoles = user.getEffectiveRoles(currentLocation)
-            if (!userRoles.any { Role role -> role.roleType == accessRule?.accessRules?.minimumRequiredRole }) {
+            def userService = ApplicationHolder.getApplication().getMainContext().getBean("userService")
```

Comment:
```
ApplicationHolder.application.mainContext.getBean("userService")
```

Reply:
This doesn't feel right so we should add this to our tech debt pile to figure out a better way to centralize this type of custom authorization logic.

Reply:
@jmiranda what would you suggest at this point? I believe this one has to go to release and we already do have similar things implemented. Should we consider pushing it further and then refactor all this stuff after the tech debt or shouldn't we merge this one?

Reply:
@kchelstowski Yeah I meant we should move forward with this but make sure we add this to our tech debt to discuss it further.

Reply:
@kchelstowski "This doesn't feel right" was in reference to the use of the service in the StockMovement class, not my recommended change. I'd still like you to make the change suggested above.